lib/conjur/feature_flags/features.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Conjur
+  module FeatureFlags
+    # TODO: Add ability to turn all flags on by default (to insure tests can) be run against flagged features
+    class Features
+      # Instantiates a new Conjur::FeatureFlags::Features object
+      #
+      # @param logger [Logger] the logger used for output
+      # @param environment_settings [Hash] the environment variable has (typically ENV)
+      # @param feature_flags [Array<Symbol>] the list of valid feature flags
+      #
+      # @return [Conjur::FeatureFlags::Features]
+      def initialize(logger:, environment_settings:, feature_flags: [])
+        @logger = logger
+        @enabled_features = merge_with_environment_variable_flags(
+          features: feature_flags,
+          environment: environment_settings
+        )
+      end
+
+      # Check to see if a feature flag has been enabled
+      #
+      # @param feature_name [Symbol] the feature flag we want to check
+      #
+      # @return [Boolean]
+      def enabled?(feature_name)
+        @logger.debug("Conjur::FeatureFlags::Features#enabled? - feature flag name: '#{feature_name}'")
+        result = @enabled_features.key?(feature_name.downcase.to_sym) &&
```

Comment:
This method expects a symbol.  It should just error if it's not a symbol, or else just assume `feature_name` is symbol.  

That is, it's our own code calling this, so it should already be valid.  If `feature_name` is coming from user input at some point, the client code handling that should do the validation, rather than us providing the service as a courtesy here.

Reply:
I updated this method to raise an error if we get anything but a symbol.